chat-plugins/datasearch.js

```diff
@@ -1382,7 +1382,8 @@ function runItemsearch(target, cmd, canAll, message) {
 		}
 
 		// iterate over found items again to make sure they all are the best match
-		for (let [i, id] of foundItems.entries()) {
+		// use a reversed copy for iteration so we can remove items mid-loop without changing the remaining indices
```

Comment:
Actually, in hindsight, this entire loop should just be replaced with a call to `filter`.